### PR TITLE
Cleanup DWebBrowserEvents2 and add some WebBrowser tests

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup DWebBrowserEvents2
- Add some WebBrowser tests

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1958)

### CODE REVIEW

File: src/Common/src/Interop/SHDocVw/Interop.DWebBrowserEvents2.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+            [DispId(250)]
+            void BeforeNavigate2(
+                [MarshalAs(UnmanagedType.IDispatch)] object pDisp,
+                ref object URL,
+                ref object flags,
+                ref object targetFrameName,
+                ref object postData,
+                ref object headers,
+                ref bool cancel);
```
Comment (RussKie) [inline src/Common/src/Interop/SHDocVw/Interop.DWebBrowserEvents2.cs]: Should this be `BOOL`?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PictureBoxTests.cs [hunk 2]
```diff
@@ -1530,7 +1530,7 @@ public void PictureBox_Text_SetWithHandler_CallsTextChanged()
         [CommonMemberData(nameof(CommonTestHelper.GetBoolTheoryData))]
         public void PictureBox_Visible_Set_GetReturnsExpected(bool value)
         {
-            var control = new Control
+            var control = new PictureBox
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PictureBoxTests.cs]: This change seems completely orthogonal to the main change.
Please submit is separately

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs [hunk 3]
```diff
@@ -3875,14 +3875,7 @@ public void Disconnect()
...
                         SynchronizationContext context = SynchronizationContext.Current;
-                        if (context == null)
-                        {
-                            Debug.Fail("Attempted to disconnect ConnectionPointCookie from the finalizer with no SynchronizationContext.");
-                        }
-                        else
-                        {
-                            context.Post(new SendOrPostCallback(AttemptDisconnect), null);
-                        }
+                        context?.Post(new SendOrPostCallback(AttemptDisconnect), null);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs]: ditto

File: src/Common/src/Interop/SHDocVw/Interop.DWebBrowserEvents2.cs (hunk 1 above)
Comment (hughbe) [reply src/Common/src/Interop/SHDocVw/Interop.DWebBrowserEvents2.cs]: Nope it’s VARIANT_BOOL

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs]: Nope to get the tests working we need to remove this assert

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PictureBoxTests.cs (hunk 2 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PictureBoxTests.cs]: I included this here because were adding tests for WebBrowser.Visible and updating similar tests
